fc/cli/vfrcheck.py

```diff
@@ -0,0 +1,277 @@
+    SHEET_CREATE_URL = auth.sheet_create_url()
+
+    if SERVICE_ACCOUNT_FILE is None or SHEET_CREATE_URL is None:
+        cli_library.echo('Google service account credential file path and sheet create url must be defined '
+                         'in order to generate a report file.')
+
+    cli_library.echo('retrieving all stories')
+
+    # get stories ordered by duration
+    jsonr = tasks.search_for_stories_ord_duration(auth)
+
+    cli_library.echo('story retrieval complete')
+
+    # setup and build credentials for google api calls
+    SCOPES = ['https://www.googleapis.com/auth/spreadsheets', 'https://www.googleapis.com/auth/drive.file']
+
+    credentials = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+
+    service = build('sheets', 'v4', credentials=credentials)
+
+    report_title = 'VFR Sanity Check ' + date.today().strftime('%m-%d-%y')
+
+    response = requests.get(SHEET_CREATE_URL + report_title)
+
+    response.raise_for_status()
+
+    response_json = response.json()
+
+    spreadsheet_id = None
+    folder_id = None
+
+    if response_json['success']:
+        spreadsheet_id = response_json['newSheetId']
+        folder_id = response_json['destFolderId']
+        cli_library.echo('successful spreadsheet creation, https://docs.google.com/spreadsheets/d/{}/edit#gid=0, '
+                         .format(spreadsheet_id) +
+                         'folder: {}'.format(folder_id))
+
+    # update the individual sheet name in the spreadsheet we just created
+    sheet_update = {
+        'requests': {
+            'updateSheetProperties': {
+                'properties': {
+                    'title': 'Duration'
+                },
+                'fields': 'title'
+            }
+        }
+    }
+
+    response = service.spreadsheets().batchUpdate(
+        spreadsheetId=spreadsheet_id,
+        body=sheet_update).execute()
+
+    cli_library.echo('default sheet title updated')
+
+    # create arrays of data to put into the sheet
+    # here's a header row
+    data = [{'range': 'A1:D1',
+             'values': [['Cost of Delay', 'Issue key', 'Summary', 'Duration']]}]
+    idx = 2
+
+    # create an array of requests that will hold "updateCells" requests
+    # these are rows that will have the background color changed
+    update_cells_requests = []
+
+    update_cells = {
+        'requests': update_cells_requests
+    }
+
+    # create a var to hold the "previous" duration value
+    prev_duration = 0
+
+    color_format_json = {
+                          'userEnteredFormat': {
+                              'backgroundColor': {
+                                  'red': 0,
+                                  'green': 0.4,
+                                  'blue': 0,
+                                  'alpha': 1.0
+                              }
+                          }
+                        }
+
+    cli_library.echo('generating data for duration sheet cells')
```

Comment:
The following code also seems that it can be split up into different functions.  At the very least one per sheet.